### PR TITLE
 Add `replace` option for `redirect` operator

### LINKED ISSUE #44: Add `replace` option for `redirect` operator
Like with `route.navigate`, we should also add `replace` option for `redirect`

### PR DESCRIPTION
**Pull request by velialiev**
Add replace option for redirect operator

Closes #44

### CODE REVIEW

**Comment by sergeysova** [comment]
@velialiev Thank you for contribution.
Would you like to add tests for this feature?

**Comment by velialiev** [comment]
> @velialiev Thank you for contribution.
> Would you like to add tests for this feature?

Yes, sure

**Comment by velialiev** [comment]
Would it be a good idea to separate and reuse common properties from `type RedirectParams` condition's cases in scope of this PR or should the refactoring be done separately?

There are too many duplicated properties in there and will be even more over time (see #46)

@sergeysova

File: src/methods/redirect.ts [hunk 1]
```diff
@@ -14,13 +14,15 @@ type RedirectParams<T, Params extends RouteParams> = Params extends EmptyObject
       clock?: Clock<T>;
       route: RouteInstance<Params>;
       query?: ((clock: T) => RouteQuery) | Store<RouteQuery> | RouteQuery;
+      replace?: boolean;
```
Comment (js2me) [inline src/methods/redirect.ts:17]: I think better to change this type to  
```ts
replace?: ((clock: T) => boolean) | Store<boolean> | boolean;
```

I think that's more flexible functionality   

@sergeysova what do you think ?

Comment (sergeysova) [reply src/methods/redirect.ts:17]: @js2me Makes sense

**Comment by sergeysova** [comment]
@velialiev I think it is better to create another PR with refactoring.
But before, we need to setup type-tests to be sure this refactoring won’t brake anything.

**Comment by velialiev** [comment]
Completed in #55

**Pull request closed without merging**